### DISCUSSION TITLE
How to reopen TWS on rdesktop after it is closed?

### DISCUSSION THREAD

**Discussion by jeremytanjianle** [gnzsnz/ib-gateway-docker#196] [Q&A] [answered]
Hi there,

When using the docker image while simultaneously running live and paper TWS, what if we accidentally close the paper TWS for whatever reason?

How can we re-open the paper TWS, without restarting the docker container and disrupting the live TWS instance?

Thanks and regards

**Comment by gnzsnz** [comment]
if you are into bash scripting you need to replicate the environment vars defined [here](https://github.com/gnzsnz/ib-gateway-docker/blob/eb3850ad810c90b322d8daf37ebf4288a77de176/image-files/tws-scripts/run_tws.sh#L116) and run IBC. that's not so simple

The way it works is quite simple, once the last TWS (paper or live doesn't really matters) instance exists, both will be restarted.

**Reply by jeremytanjianle** [reply]
Thanks @gnzsnz 

Thanks for pointing to the bash scripting components.

Sorry to clarify, what do you mean by "when last TWS instance exists"?

If you meant to say "when the last TWS instance closes," does that mean the solution is to close all TWS instances?

Thanks again

**Reply by gnzsnz** [reply, marked as answer]
yes correct, once the last one is closed the start script will be run again and start-up TWS again. depending on your settings this will be live, paper or both.
